stdlib/public/runtime/CMakeLists.txt

```diff
@@ -75,8 +75,34 @@ set(swift_runtime_library_compile_flags ${swift_runtime_compile_flags})
 list(APPEND swift_runtime_library_compile_flags -DswiftCore_EXPORTS)
 list(APPEND swift_runtime_library_compile_flags -I${SWIFT_SOURCE_DIR}/stdlib/include/llvm/Support -I${SWIFT_SOURCE_DIR}/include)
 
-set(sdk "${SWIFT_HOST_VARIANT_SDK}")
+if(SWIFT_${SWIFT_HOST_VARIANT_SDK}_${SWIFT_HOST_VARIANT_ARCH}_ICU_STATICLIB)
+  set(libicu_i18n_a -licui18nswift)
+  set(libicu_uc_a -licuucswift)
+  set(libicu_data_a -licudataswift)
+else()
+  find_package(ICU REQUIRED COMPONENTS uc i18n)
+  get_filename_component(ICU_UC_LIBDIR "${ICU_UC_LIBRARIES}" DIRECTORY)
+  get_filename_component(ICU_I18N_LIBDIR "${ICU_I18N_LIBRARIES}" DIRECTORY)
+
+  set(libicu_i18n_a -licui18n)
+  set(libicu_uc_a -licuuc)
+  set(libicu_data_a)
+  if(EXISTS ${ICU_I18N_LIBDIR}/libicui18n.a AND
+      EXISTS ${ICU_UC_LIBDIR}/libicuuc.a AND
+      EXISTS ${ICU_UC_LIBDIR}/libicudata.a)
+    set(libicu_i18n_a ${ICU_I18N_LIBDIR}/libicui18n.a)
+    set(libicu_uc_a ${ICU_UC_LIBDIR}/libicuuc.a)
+    set(libicu_data_a ${ICU_UC_LIBDIR}/libicudata.a)
+  endif()
+endif()
+set(libdl)
 if(SWIFT_BUILD_STATIC_STDLIB AND "${sdk}" STREQUAL "LINUX")
+  set(libdl -ldl)
```

Comment:
Ugh; I really wish that we could use `CMAKE_DL_LIBS`, unfortunately, the custom cross-compilation build system in Swift doesn't support that.  We already have the special handling for `dl` in `_add_target_variant_link_flags`, can we re-use that rather than duplicate knowledge of libdl here?  Alternatively, perhaps we should consider creating a `SWIFT_DL_LIBS_${SWIFT_HOST_VARIANT_SDK}_${SWIFT_HOST_VARIANT_ARCH}` variable to indicate if/when libdl is needed on a platform?

Reply:
This one's a little more complex, since it appears (to me) that _add_target_variant_link_flags works with a CMake target, and maybe there's a way to create a dummy target and pull the link libraries. The latter is probably simpler, but I'll probably need to spin up a Linux build to do this properly.

Added a TODO and I'll take a look at this in a later PR. Does that sound reasonable?